### PR TITLE
Feature/language handling topics / forums

### PR DESCRIPTION
**Pull request by macjohnny**
this ignores sys_language_uid when listing topics & forums, since this should be language independent. otherwise, a topic created in a certain language will not be visible for users with a different language. language separation should rather be achieved by having different forums for different languages.

we could discuss whether this is the right place to implement this. 
i also tried implementing a function `initializeObject()` and set the `$this->setDefaultQuerySettings()`, however, `initializeObject()` is  never called.
See also https://wiki.typo3.org/Default_Orderings_and_Query_Settings_in_Repository

### CODE REVIEW

**Comment by macjohnny** [comment]
@martin-helmich what do you think?

**Comment by macjohnny** [comment]
by the way: setting the query settings in the constructor did not work either.
concerning the `initializeObject()` method, see also https://forum.typo3.org/index.php?t=msg&goto=529999&

File: Classes/Domain/Repository/Forum/ForumRepository.php [hunk 1]
```diff
@@ -34,13 +34,34 @@
...
+     * @api
+     */
+    public function createQuery() {
+        $query = $this->persistenceManager->createQueryForType($this->objectType);
+        if ($this->defaultOrderings !== array()) {
+            $query->setOrderings($this->defaultOrderings);
+        }
+        if ($this->defaultQuerySettings !== NULL) {
+            $query->setQuerySettings(clone $this->defaultQuerySettings);
+        }
```
Comment (martin-helmich) [inline Classes/Domain/Repository/Forum/ForumRepository.php]: This should be a bit more forward-compatible:

```
$query = parent::createQuery();
```

File: Classes/Domain/Repository/Forum/ForumRepository.php [hunk 2]
```diff
@@ -34,13 +34,34 @@
...
+
+        // don't add sys_language_uid constraint
+        $query->getQuerySettings()->setRespectSysLanguage(FALSE);
```
Comment (martin-helmich) [inline Classes/Domain/Repository/Forum/ForumRepository.php]: I'm not sure if this is backwards compatible. Maybe there should be some mechanism (TS switch?) to opt-in for this?

File: Classes/Domain/Repository/Forum/TopicRepository.php [hunk 3]
```diff
@@ -35,6 +35,27 @@
...
+        }
+        if ($this->defaultQuerySettings !== NULL) {
+            $query->setQuerySettings(clone $this->defaultQuerySettings);
+        }
+
+        // don't add sys_language_uid constraint
+        $query->getQuerySettings()->setRespectSysLanguage(FALSE);
+
+        return $query;
+    }
```
Comment (martin-helmich) [inline Classes/Domain/Repository/Forum/TopicRepository.php]: see above

File: Classes/Domain/Repository/Forum/ForumRepository.php (hunk 1 above)
Comment (macjohnny) [inline Classes/Domain/Repository/Forum/ForumRepository.php]: you are right, i updated the code

File: Classes/Domain/Repository/Forum/ForumRepository.php (hunk 2 above)
Comment (macjohnny) [inline Classes/Domain/Repository/Forum/ForumRepository.php]: well, it clearly isn't, but i can hardly imagine someone would use it as a multi-language forum with the current translation handling. quite a few features don't work correctly in multi-lingual mode without the fixes i provided in the other pull requests for the ajax dispatcher.
so i would consider it ok to break backwards-compatibility here.

File: Classes/Domain/Repository/Forum/TopicRepository.php (hunk 3 above)
Comment (macjohnny) [inline Classes/Domain/Repository/Forum/TopicRepository.php]: see ForumRepository

**Pull request merged**